pkg/server/spec/pod.go

```diff
@@ -10,70 +10,72 @@ const (
 	slugVolumeMountPath = "/slug"
 )
 
-type ContainerLimits struct {
-	CPU    string
-	Memory string
-}
-
-type VolumeMounts struct {
-	Name      string
-	MountPath string
-	ReadOnly  bool
-}
-
 type Volume struct {
-	Name       string
-	SecretName string
-	EmptyDir   bool
-}
-
-type Container struct {
-	Name            string
-	Namespace       string
-	Image           string
-	ContainerLimits *ContainerLimits
-	Env             map[string]string
-	VolumeMounts    []*VolumeMounts
-	Args            []string
+	Name          string
+	SecretName    string
+	ConfigMapName string
+	EmptyDir      bool
 }
 
 type Pod struct {
-	Container
+	Name           string
+	Namespace      string
+	Containers     []*Container
 	Volumes        []*Volume
 	InitContainers []*Container
 }
 
-func NewPod(name, image string, a *app.App, envVars map[string]string, fs storage.Storage) *Pod {
-	ps := &Pod{
-		Container: Container{
-			Name:      name,
-			Namespace: a.Name,
-			Image:     image,
-			Env:       envVars,
+func newPodVolumes(appName string, fs storage.Storage, hasNginx bool) []*Volume {
+	volumes := []*Volume{
+		{
+			Name:       "storage-keys",
+			SecretName: fs.K8sSecretName(),
 		},
-		Volumes: []*Volume{
-			{
-				Name:       "storage-keys",
-				SecretName: fs.K8sSecretName(),
-			},
-			{
-				Name:     slugVolumeName,
-				EmptyDir: true,
-			},
+		{
+			Name:     slugVolumeName,
+			EmptyDir: true,
 		},
 	}
+	if hasNginx {
+		volumes = append(volumes, &Volume{
+			Name:          "nginx-conf",
+			ConfigMapName: appName,
+		})
+	}
+	return volumes
+}
+
+func newPodContainers(name, appImage, nginxImage string, envVars map[string]string, hasNginx bool) []*Container {
+	c := []*Container{
+		newAppContainer(name, appImage, envVars, hasNginx),
+	}
+	if hasNginx {
+		c = append(c, newNginxContainer(nginxImage))
+	}
+	return c
+}
+
+func NewPod(name, nginxImage, image string, a *app.App, envVars map[string]string, fs storage.Storage, hasNginx bool) *Pod {
```

Comment:
Maybe you can change args to get struct `Images` struct and check the flag `hasNginx` with `imgs.NginxImage` field.

Reply:
Make senses